packages/salesforcedx-vscode-core/src/orgBrowser/metadataCmp.ts

```diff
@@ -148,11 +140,14 @@ export function buildComponentsList(
   metadataType: string
 ): string[] {
   try {
-    const fileData = JSON.parse(fs.readFileSync(componentsPath, 'utf8'));
     const metaComponents = [];
-    for (const component of fileData) {
-      if (!isNullOrUndefined(component.fullName)) {
-        metaComponents.push(component.fullName);
+    const fileData = fs.readFileSync(componentsPath, 'utf8');
+    if (fileData !== 'undefined') {
+      const fileObject = JSON.parse(fileData);
+      for (const component of fileObject) {
```

Comment:
Due to the nature of the api, if the response only returns one item fileObject will be an Object instead of a List. The side effect is that you'll be iterating over the keys of the object and not the object itself. This should be handled here